Demo

JanMouwes opened the pull request.

JanMouwes committed:
Merge pull request #12 from christiaanko/dev

Dev

Christiaan-Kornaat committed:
Merge pull request #13 from christiaanko/dev

Added some styling

JanMouwes committed:
Search works, untangled some code, added various services. Added Adaguc-url-builder

JanMouwes committed:
Rebased commit: Search works, untangled some code, added various services. Added Adaguc-url-builder

JanMouwes committed:
Rebased commit: Search works, untangled some code, added various services. Added Adaguc-url-builder

JanMouwes committed:
Rebased commit: Search works, untangled some code, added various services. Added Adaguc-url-builder

JanMouwes committed:
Fixed various TS-strict errors

JanMouwes committed:
Fixed various TS-strict errors

JanMouwes committed:
Fixed various TS-strict errors

JanMouwes committed:
- Added catalog-item-model-parsing
- Added string-unit-tests
- Added trimCharsLeft, trimCharsRight and trimChars
- Fixed Jest-unit-testing

JanMouwes committed:
- Added catalog-item-model-parsing
- Added string-unit-tests
- Added trimCharsLeft, trimCharsRight and trimChars
- Fixed Jest-unit-testing

JanMouwes committed:
- Added catalog-item-model-parsing
- Added string-unit-tests
- Added trimCharsLeft, trimCharsRight and trimChars
- Fixed Jest-unit-testing

JanMouwes committed:
- Added catalog-item-model-parsing
- Added string-unit-tests
- Added trimCharsLeft, trimCharsRight and trimChars
- Fixed Jest-unit-testing

JanMouwes committed:
Merge branch 'feature-adaguc-integration' of https://github.com/christiaanko/ESGF_Portal_React into feature-adaguc-integration

JanMouwes committed:
- Removed deprecated searchResultProvider from qf- & xpf-wrapper
- Refactored ResultItem & ResultItemWrapper to work with asynchronous loading of data
- Searching now works
- Added EsgfSearchManager & EsgfSearchQuery for ESGF-searching
- Fixed ESGFSearchResultsProvider to actually work & co-operate with the service
- ESGFSearchResultsProvider now works with EsgfSearchQuery

JanMouwes committed:
- Removed deprecated searchResultProvider from qf- & xpf-wrapper
- Refactored ResultItem & ResultItemWrapper to work with asynchronous loading of data
- Searching now works
- Added EsgfSearchManager & EsgfSearchQuery for ESGF-searching
- Fixed ESGFSearchResultsProvider to actually work & co-operate with the service
- ESGFSearchResultsProvider now works with EsgfSearchQuery

marijnkoel committed:
inlineSpinner

marijnkoel committed:
Rebased commit inlineSpinner

JanMouwes committed:
- Fixed: ResultItem actually updates
- Added: Result-count in ResultWrapper
- Removed some redundant code
- Fixed event spam when sending multiple requests
- Removed redundant debug-logging

JanMouwes committed:
- Fixed: ResultItem actually updates
- Added: Result-count in ResultWrapper
- Removed some redundant code
- Fixed event spam when sending multiple requests
- Removed redundant debug-logging

JanMouwes committed:
Merge branch 'feature-adaguc-integration' of https://github.com/christiaanko/ESGF_Portal_React into feature-adaguc-integration

# Conflicts:
#	src/js/components/results-search/result-wrapper/result-wrapper.component.tsx

Also, added spinner when loading new results

Christiaan-Kornaat committed:
cleaned up xpf wrapper and added the qfc file

JanMouwes committed:
Rebased commit:
- Removed filters with one item or less
- Fixed searching: now searches for subsets as well
- Removed forced-uppercasing
- Deprecated ESGFPropertyFormatter
- Added EsgfFilterPropertyVM for displaying filterProperties
- Centralized Searchers; added Searcher (central class for all other searchers to use
- Added ignoreCaseComparator
- Added facets.map.json; to be used for an interface that allows easy mapping of facets & properties

JanMouwes committed:
- Removed filters with one item or less
- Fixed searching: now searches for subsets as well
- Removed forced-uppercasing
- Deprecated ESGFPropertyFormatter
- Added EsgfFilterPropertyVM for displaying filterProperties
- Centralized Searchers; added Searcher (central class for all other searchers to use
- Added ignoreCaseComparator
- Added facets.map.json; to be used for an interface that allows easy mapping of facets & properties

JanMouwes committed:
Merged & rebased with dev (finished)

JanMouwes committed:
Fixed search

Christiaan-Kornaat committed:
moved logical components from xpf-wrapper to app component (xpf not working)

JanMouwes committed:
Fixed some UnorderedList rerenders

JanMouwes committed:
Fixed some UnorderedList rerenders

Christiaan-Kornaat committed:
Merge branch 'feature-adaguc-integration' into QFManager

JanMouwes committed:
Removed some redundant code, changed some outdated code to use newer techniques

marijnkoel committed:
refactored qftiles to tiles and added tilefactory

marijnkoel committed:
refactored qftiles to tiles and added tilefactory

marijnkoel committed:
refactored qftiles to tiles and added tilefactory

marijnkoel committed:
Rebased: refactored qftiles to tiles and added tilefactory

Christiaan-Kornaat committed:
Rebased: created overlayFactory and overlay component

Christiaan-Kornaat committed:
created overlayFactory and overlay component

Christiaan-Kornaat committed:
created overlayFactory and overlay component

Christiaan-Kornaat committed:
created overlayFactory and overlay component

marijnkoel committed:
overlay + qfc wrapper done

marijnkoel committed:
Rebased: overlay + qfc wrapper done

marijnkoel committed:
overlay + qfc wrapper done

marijnkoel committed:
overlay + qfc wrapper done

marijnkoel committed:
Update qfc-wrapper.component.tsx

marijnkoel committed:
Update qfc-wrapper.component.tsx

marijnkoel committed:
Update qfc-wrapper.component.tsx

marijnkoel committed:
Rebased: Update qfc-wrapper.component.tsx

JanMouwes committed:
- Generified XPF-wrapper - partly

marijnkoel committed:
progress of the day  iguess

marijnkoel committed:
Rebased: progress of the day  iguess

marijnkoel committed:
progress of the day  iguess

marijnkoel committed:
progress of the day  iguess

JanMouwes committed:
- Completely rewrote the XPF-Wrapper to be more generic and clean
- Fixed various bugs that emerged from the generified XPFWrapper
- Removed various deprecated classes
- Results now show on initial load
- HtmlList now updates when fed the same list-reference
- Added ColumnedPage (and columns themselves & their factories) for creating columned pages
- Rewrote EventEmitter to TypeScript
- Added InfoTabVmFactory
-

marijnkoel committed:
local storage + default

marijnkoel committed:
Rebased: local storage + default

marijnkoel committed:
local storage + default

marijnkoel committed:
local storage + default

marijnkoel committed:
Merge branch 'customizer-qf' of https://github.com/christiaanko/ESGF_Portal_React into customizer-qf

JanMouwes committed:
- Fixed various errors
- Fixed bug where XPF would select its info-screen when it wasn't supposed to

marijnkoel committed:
Update app.tsx

marijnkoel committed:
Update tileController-local.ts

JanMouwes committed:
Merge branch 'customizer-qf' of https://github.com/christiaanko/ESGF_Portal_React into customizer-qf

# Conflicts:
#	src/js/components/quick-filter-customizer/wrapper/qfc-wrapper.component.tsx
#	src/js/components/quick-filter-search/wrapper/qf-wrapper.component.tsx
#	src/js/controllers/localstorage/tiles/tileController-local.ts
#	src/style/ESGFSearch.scss

marijnkoel committed:
Merge pull request #15 from christiaanko/customizer-qf

Customizer qf

marijnkoel committed:
content tab ding geval

JanMouwes committed:
- Added QFC-Customizer & functionality
- Moved QFListItemFactory to ListItemFactoryFactory
- Minor code changes

JanMouwes committed:
Merge branch 'customizer-qf' of https://github.com/christiaanko/ESGF_Portal_React into customizer-qf

JanMouwes committed:
- Added Customizer-functionality

JanMouwes committed:
- Added full Customizer-functionality: delete & save
- Added Success & Danger buttons

JanMouwes committed:
- Minor code cleanliness fixes
- Minor site content changes
- Max 8 tiles are now allowed

JanMouwes committed:
- Disabled tab-animations

marijnkoel committed:
styling

marijnkoel committed:
Merge branch 'customizer-qf' of https://github.com/christiaanko/ESGF_Portal_React into customizer-qf

marijnkoel committed:
styling enzo

marijnkoel committed:
Update qfc-wrapper.component.tsx

Christiaan-Kornaat committed:
Removed tabs and added a back button to switch between customizer pages

Christiaan-Kornaat committed:
Merge branch 'customizer-qf' of https://github.com/christiaanko/ESGF_Portal_React into customizer-qf

marijnkoel committed:
remove item from tile

marijnkoel committed:
Merge branch 'customizer-qf' of https://github.com/christiaanko/ESGF_Portal_React into customizer-qf

marijnkoel committed:
Added a unmount so wf updates again

marijnkoel committed:
Merge pull request #16 from christiaanko/customizer-qf

Customizer qf

JanMouwes committed:
- Minor code changes: added types to methods, method-parameters and attributes

JanMouwes committed:
Merge branch 'feature-adaguc-integration' of https://github.com/christiaanko/ESGF_Portal_React into feature-adaguc-integration

# Conflicts:
#	src/js/components/shared/list-html/list-html.component.tsx

JanMouwes committed:
- Fixed minor issue that prevented compilation

Christiaan-Kornaat committed:
Added propertyDTO

marijnkoel committed:
converter + controller done

Christiaan-Kornaat committed:
added a list item factory

Christiaan-Kornaat committed:
Merge branch 'presets' of https://github.com/christiaanko/ESGF_Portal_React into presets

marijnkoel committed:
started on wrapper

marijnkoel committed:
progress

marijnkoel committed:
fixed jan's grammar

marijnkoel committed:
save cuz end of day

marijnkoel committed:
generic local storage to replace tile and preset controllers

marijnkoel committed:
preset doet dingen

Christiaan-Kornaat committed:
changed selectedfilter to selectedfilterproperties

marijnkoel committed:
Changed edit button and such

JanMouwes committed:
- Fixed issue where unit tests wouldn't run

marijnkoel committed:
SOme edits on presets save almost done

JanMouwes committed:
- Re-arranged data-file-structure: files are now organised by theme rather than by Class-type.
- Results are now fetched and displayed when result-list-item is clicked
- Added ResultIcons.Download component
- Added bootstrapped ResultItem
- Added ICatalogProvider and -Service and their implementation classes
- SearchService now no longer searches twice in parallel when asked twice
- Added IAdagucUrlBuilder.buildCatalogUrl() - does what it says on the tin
- Reworked EsgfCatalogItemDTO to reflect actual retrieved results
- Removed a whole bunch of redundant code

JanMouwes committed:
Merge branch 'feature-adaguc-integration' of https://github.com/christiaanko/ESGF_Portal_React into feature-adaguc-integration

# Conflicts:
#	src/js/components/shared/list-html/list-html.component.tsx

JanMouwes committed:
- Fixed download link

JanMouwes committed:
- Result items look a little better

JanMouwes committed:
- Added No connection-icon to QfWrapper

JanMouwes committed:
- Changed the TileController to return a Promise which returns multiple Tiles, rather than multiple Promises that return one tile
- EsgfFilterProvider no longer caches failed requests
- Added QfTileConverter-unit test

JanMouwes committed:
Merge Presets-branche with Concept-branche

JanMouwes committed:
Merge pull request #17 from christiaanko/presets

Presets

marijnkoel committed:
Delete tileController-local.ts

JanMouwes committed:
- Fixed issue where index would show NaN

marijnkoel committed:
lots of changes on presets

marijnkoel committed:
repaired qfc

marijnkoel committed:
herokutest

marijnkoel committed:
Save delete new all works now

marijnkoel committed:
Update xpf-wrapper.component.tsx

JanMouwes committed:
- Added result-paging (added PageSelector)
- Reformed Icons

marijnkoel committed:
Merge branch 'feature-adaguc-integration' into demo

Christiaan-Kornaat committed:
styled button

marijnkoel committed:
Update package.json

marijnkoel committed:
Create server.js

marijnkoel committed:
Update ESGFSearch.scss

Christiaan-Kornaat committed:
ggsjd

marijnkoel committed:
Merge branch 'presets' into demo

JanMouwes committed:
- Fixed Buttons.Options
- Changed Icon & FontAwesomeIcon to be more semantically fitting

marijnkoel committed:
Merge branch 'feature-adaguc-integration' into demo

marijnkoel committed:
Update app.tsx

marijnkoel commented on src/js/components/quick-filter-customizer/wrapper/qfc-wrapper.component.tsx:
```diff
@@ -0,0 +1,244 @@
+        let {filterProvider} = props;
+        this._filterProvider = filterProvider;
+
+        let defaultTiles = [
```
disgusting

marijnkoel replied:
not even needed in the qfc?

marijnkoel commented on src/js/components/quick-filter-search/wrapper/qf-wrapper.component.tsx:
```diff
@@ -1,39 +1,148 @@
         this._quickFilterProvider = qfProvider;
+        this._filterProvider = filterProvider;
+
+        let defaultTiles = [
```
another one

JanMouwes replied:
stop it

JanMouwes commented on src/js/components/expanded-property-finder/wrapper/xpf-wrapper.component.tsx:
```diff
@@ -0,0 +1,353 @@
+        this.setState({ presetsListItems: presets });
+    }
+
+    handleBackClick() {
```
inaccurate method naming

JanMouwes commented on src/js/components/expanded-property-finder/wrapper/xpf-wrapper.component.tsx:
```diff
@@ -0,0 +1,353 @@
+
+    async savePreset() {
+        let presets = this.state.presetsListItems;
+        this._presetController.setLocalstorage(presets);
```
I feel like a data layer between localStorageController and client is required. PresetDAL?

Christiaan-Kornaat replied:
dat hadden we eerst maar dan krijg je voor alles wat met de local storage interacteerd een losse DAL die exact hetzelfde is. Dus dat is niet handig

JanMouwes commented on src/js/components/expanded-property-finder/wrapper/xpf-wrapper.component.tsx:
```diff
@@ -0,0 +1,353 @@
+
+        // let createSelectColumnTab = columnName => newTab => this.selectTab(columnName, newTab);
+
+        this.state.className += " XPF-Wrapper";
```
Keeps adding classnames 
![Schermafbeelding 2019-06-07 om 14 28 53](https://user-images.githubusercontent.com/13809326/59104060-c949c000-8930-11e9-803d-1695b6b9e7df.png)

JanMouwes replied:
note: this doesn't break anything, but to say it's supposed to be like that...

Christiaan-Kornaat replied:
Geen idee waarom dat uberhaupt op die manier wordt gezet

JanMouwes commented on src/js/components/presets/column/preset-column.component.tsx:
```diff
@@ -0,0 +1,91 @@
+        return (
+            <div className="content-tab-preset-customizer-wrapper">
+                <div className="preview-presets">
+                    {this.state.preset.properties != null ?
```
liever in een apart variabele

JanMouwes commented on src/js/components/presets/wrapper/preset-customizer-wrapper.tsx:
```diff
@@ -0,0 +1,235 @@
+        this._selectedPropertyManager.events.selectionChanged.subscribe(() => this.update());
+
+        this._searchFunctions = {
+            filters: new ESGFFilterSearcher().search,
```
move to Searchers.ESGF-namespace, then rename to FilterSearcher?

JanMouwes commented on src/js/components/quick-filter-customizer/column/qfc-content-tab-preview.component.tsx:
```diff
@@ -0,0 +1,102 @@
+        this.handlePropertiesChange(properties);
+    }
+
+    handlePropertiesChange(properties) {
```
Missing type annotation

JanMouwes commented on src/js/components/quick-filter-customizer/column/qfc-content-tab-preview.component.tsx:
```diff
@@ -0,0 +1,102 @@
+        this.saveTile(qfTile);
+    }
+
+    saveTile(tile) {
```
Missing type annotation

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-customizer-wrapper.component.tsx:
```diff
@@ -0,0 +1,236 @@
+
+        let createPropertyInfoTab = (property: ESGFFilterPropertyDTO) => columnTabFactory.createInfoTab(infoTabVMFactory.createPropertyVM(property));
+
+        let showInfo = async (columnTab: JSX.Element) => this.setTab(ColumnPosition.Right, columnTab, "property-info");
```
why is "property-info" still hardcoded?

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-customizer-wrapper.component.tsx:
```diff
@@ -0,0 +1,236 @@
+            selectTab(ColumnPosition.Right, infoTabName);
+        };
+
+        let createPropertyInfoTab = (property: ESGFFilterPropertyDTO) => columnTabFactory.createInfoTab(infoTabVMFactory.createPropertyVM(property));
```
InfoTabs-namespace, PropertyInfoTab-class

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-customizer-wrapper.component.tsx:
```diff
@@ -0,0 +1,236 @@
+                                                headerButtons: [optionComponents.filters]
+                                            }}/>;
+
+        let PropertyList = <PageColumnListTab title={"Properties"}
```
move these to a central namespace, to not use the lib directly in the app

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-wrapper.component.tsx:
```diff
@@ -0,0 +1,244 @@
+        let {filterProvider} = props;
+        this._filterProvider = filterProvider;
+
+        let defaultTiles = [
```
move these to a different place, seriously.

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-wrapper.component.tsx:
```diff
@@ -0,0 +1,244 @@
+    currentCustomTile,
+} & ColumnedPageProps;
+
+export default class QFCWrapper extends Component<QFCProps> {
```
move to QuickFilterCustomiser-namespace, just name it Wrapper

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-wrapper.component.tsx:
```diff
@@ -0,0 +1,244 @@
+        this.update();
+    }
+
+    openCustomiser(qfTile) {
```
Add type annotation

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-wrapper.component.tsx:
```diff
@@ -0,0 +1,244 @@
+        window.alert("tile saved");
+    }
+
+    async saveTile(tile) {
```
Add type annotation

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-wrapper.component.tsx:
```diff
@@ -0,0 +1,244 @@
+        });
+    }
+
+    deleteTile(tile) {
```
Add type annotation

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-wrapper.component.tsx:
```diff
@@ -0,0 +1,244 @@
+        let hasMaxTiles = qfTiles.length >= 8;
+        let tileFactory = new TileFactory();
+
+        let iconTileAdd = new QFFilterTileDTO("Add Quick Filter", "#3f3f3f", "fas fa-plus-circle", []);
```
Move to Tiles-namespace

JanMouwes commented on src/js/components/quick-filter-search/esgf-qfilter-tile/qf-icon-tile.component.tsx:
```diff
@@ -0,0 +1,49 @@
+import React, { Component } from "react";
+import { QFFilterTileDTO } from "../../../model/dto/qf-filter-tile.dto";
+
+export class IconTile extends Component<{ QFFilterTileDTO: QFFilterTileDTO, onClick: any }> {
```
Why no central 'Tile' class? Should probably be added

JanMouwes replied:
Also, namespaces are important. Tiles.IconTile, Tiles.QuickFilterTile, etc

JanMouwes commented on src/js/components/quick-filter-search/wrapper/qf-wrapper.component.tsx:
```diff
@@ -1,39 +1,148 @@
+import {QFFilterTileJSONDTO, QFTileConverter} from "../../../data/converters/qf-tile-converter";
+import {ESGFFilterProvider} from "../../../data/esgf-filter/esgf-filter.provider";
 
+enum ErrorState {
```
Should be standardized

JanMouwes commented on src/js/components/quick-filter-search/wrapper/qf-wrapper.component.tsx:
```diff
@@ -1,39 +1,148 @@
+}
 
-export class QFWrapper extends Component<{ searchResultProvider, selectionManager, qfManager, qfProvider }> {
+export class QFWrapper extends Component<{ selectionManager: any, filterProvider: any, qfProvider: any }> {
```
Add type annotation

JanMouwes commented on src/js/components/results-search/result-items/result-dataset-list-item.component.tsx:
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import {CatalogItemDataset} from "../../../model/dto/esgf-catalog-item.dto";
+
+export default function DatasetListItem(props: { dataset: CatalogItemDataset, index: number }) {
```
move to ListItem-namespace

JanMouwes replied:
actually, no. Rename to DatasetTableRow, for it is not a List Item

JanMouwes replied:
Also, move to Results-namespace

JanMouwes commented on src/js/components/results-search/result-items/result-item.component.tsx:
```diff
@@ -0,0 +1,145 @@
+                         onOpening={handleOpen}
+                         onClosing={closeArrow}>
+                <form className={"form-inline"}>
+                    <b>View mode:</b><br/> <ViewModeSelector options={new Map([
```
move selector to separate variable for readability

JanMouwes commented on src/js/components/shared/buttons/buttons.component.js:
```diff
@@ -1,15 +1,67 @@
+ * @return {*}
+ * @constructor
+ */
+Buttons.Outline.Primary = ({title, onClick}) =>
```
A lot of Buttons-classes are very similar. Change to Central Button-class

marijnkoel commented on src/js/components/shared/pages/page-columned/page-column.component.tsx:
```diff
@@ -0,0 +1,77 @@
+     *
+     * @return Component
+     */
+    private createTab(title: string, element: JSX.Element) {
```
Move to factory?

JanMouwes commented on src/js/components/shared/overlay/overlay.component.tsx:
```diff
@@ -0,0 +1,40 @@
+        };
+    }
+
+    componentWillReceiveProps({ background, foreground }) {
```
Don't use unfiltered props-to-state-methods. access these though this.props

JanMouwes replied:
read: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

JanMouwes commented on src/js/components/shared/pages/page-columned/page-column-list-tab.component.tsx:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    componentWillReceiveProps({items, isLoading, sortFunction}: PageColumnListTabProps<TItem>) {
+        let newState: any = {
```
No using unfiltered props-to-state! https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

JanMouwes commented on src/js/components/shared/pages/page-columned/page-column-list-tab.component.tsx:
```diff
@@ -0,0 +1,95 @@
+    /**
+     * @param {string} query
+     */
+    handleSearch(query) { this.setState({searchQuery: query}); }
```
Add type annotation

JanMouwes commented on src/js/components/shared/pages/page-columned/page-column.component.tsx:
```diff
@@ -0,0 +1,77 @@
+        };
+    }
+
+    componentWillReceiveProps({tabs, activeTab}: PageColumnProps) {
```
Unfiltered props-to-state

JanMouwes commented on src/js/data/adaguc-url.builder.interface.ts:
```diff
@@ -0,0 +1,9 @@
+import ESGFDataNodeResultDTO from "../model/dto/esgf-data-node-result.dto";
+import {PageInfo} from "./adaguc-url.builder";
+
+export default interface IAdagucUrlBuilder {
```
Misleading name, would suggest Builder-pattern. Change to Factory?

JanMouwes commented on src/js/components/shared/list-html/list-html.component.tsx:
```diff
@@ -0,0 +1,58 @@
+
+        let {items, createListItem} = props;
+
+        const TAG_WHITELIST = ["ol", "ul"];
```
Use TS type-annotation:
`type HtmlListTagName = "ol" | "ul"`

JanMouwes commented on src/js/components/shared/list-html/list-html.component.tsx:
```diff
@@ -0,0 +1,58 @@
+     *
+     * @param {Array}items
+     */
+    componentWillReceiveProps({items}) {
```
Unfiltered props to state, prefer to use props directly

JanMouwes commented on src/js/components/shared/overlay/overlays.component.tsx:
```diff
@@ -0,0 +1,12 @@
+
+namespace Overlays {
+
+    export namespace QFTiles { 
```
Maybe change to tiles?

JanMouwes commented on src/js/components/expanded-property-finder/wrapper/xpf-wrapper.component.tsx:
```diff
@@ -0,0 +1,353 @@
+             addPreset(preset)();
+        }
+
+        let createPropertyInfoTab = (property: ESGFFilterPropertyDTO) => columnTabFactory.createInfoTab(infoTabVMFactory.createPropertyVM(property));
```
this whole method is a mess

JanMouwes commented on src/js/components/presets/column/preset-column.component.tsx:
```diff
@@ -0,0 +1,91 @@
+        this.handlePropertiesChange(properties);
+    }
+
+    handlePropertiesChange(properties) {
```
This method name is misleading *and* meaningless. I suggest either updatePresetProperties or just updateProperties

JanMouwes replied:
Also, type your parameters

JanMouwes commented on src/js/components/presets/wrapper/preset-customizer-wrapper.tsx:
```diff
@@ -0,0 +1,235 @@
+    private createSortState(): Map<SortState, SorterManager> {
+        let sorterFactoryFactory = new SorterFactoryFactory();
+        let createSortState = (defaultAscending, comparator) =>
+            new SorterManager(new Map([["A-Z", sorterFactoryFactory.createSorterFactory(comparator)]]), "A-Z");
```
Un-hardcode this, put it in a config-file

JanMouwes commented on src/js/components/presets/wrapper/preset-customizer-wrapper.tsx:
```diff
@@ -0,0 +1,235 @@
+        ]);
+    }
+
+    private initOptionComponents() {
```
This method is a mess.

JanMouwes commented on src/js/components/presets/wrapper/preset-customizer-wrapper.tsx:
```diff
@@ -0,0 +1,235 @@
+        this.state.columns.get(ColumnPosition.Centre).tabs.set("Properties", PropertyList);
+        this.state.columns.get(ColumnPosition.Right).tabs.set("Preset", PresetTab);
+
+        this.state.className += " XPF-Wrapper";
```
Every time this method is called, the className gets longer

JanMouwes commented on src/js/components/presets/wrapper/preset-customizer-wrapper.tsx:
```diff
@@ -0,0 +1,235 @@
+            .then(filters => this.setState({ filters: filters }));
+    }
+
+    render() {
```
This method is also a mess. Split off and clean up!

JanMouwes commented on src/js/components/presets/wrapper/preset-customizer-wrapper.tsx:
```diff
@@ -0,0 +1,235 @@
+        return super.render();
+    }
+
+    createInvertSort(columnName: SortState): () => void {
```
Why is this at the bottom?

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-customizer-wrapper.component.tsx:
```diff
@@ -0,0 +1,236 @@
+
+type ListItemFactory<TItem> = (item: TItem) => JSX.Element;
+
+export default class QfcCustomizerWrapper extends ColumnedPage<QfcCustomiserProps> {
```
This whole class looks way too much like the XpfWrapper. Use single class then Strategy-pattern to compact (https://www.youtube.com/watch?v=v9ejT8FO-7I)

JanMouwes commented on src/js/components/quick-filter-customizer/wrapper/qfc-customizer-wrapper.component.tsx:
```diff
@@ -0,0 +1,236 @@
+        let {isSelected, toggle: toggleSelected} = this._selectedPropertyManager;
+
+        let listItemFactory = new ListItemFactoryFactory();
+        let infoTabVMFactory = new InfoTabVmFactory();
```
Probably unnecessary, unless dependency-injected (this counts for all of these factories)

JanMouwes commented on src/js/components/quick-filter-search/esgf-qfilter-tile/qf-icon-tile.component.tsx:
```diff
@@ -0,0 +1,49 @@
+            <div className="qf-tile" onClick={onClick}>
+
+                <div className="qf-tile-header" style={style}><i className={icon} /> {title}</div>
+                {
```
?

JanMouwes commented on src/js/components/quick-filter-search/esgf-qfilter-tile/qf-tile.component.tsx:
```diff
@@ -0,0 +1,49 @@
+        };
+    }
+
+    componentWillReceiveProps({ QFFilterTileDTO }) {
```
destructure me!

JanMouwes commented on src/js/components/results-search/result-wrapper/result-wrapper.component.tsx:
```diff
@@ -0,0 +1,79 @@
+        this.setState(() => ({isLoading: isLoading}));
+    }
+
+    setResults(searchResult): void {
```
Type me!

JanMouwes commented on src/js/components/shared/buttons/buttons.component.js:
```diff
@@ -1,15 +1,67 @@
 import ArrowIcons from "../icons/arrow-icons.component";
 
-const Buttons = {};
+const Buttons = {Outline: {}};
```
TS me then namespace me!

JanMouwes commented on src/js/components/shared/icons/arrow-icons.component.js:
```diff
@@ -2,8 +2,19 @@ import React from "react";
 
 const ArrowIcons = {};
```
TS me then namespace me!

JanMouwes commented on src/js/components/shared/icons/icons.component.tsx:
```diff
@@ -0,0 +1,17 @@
+                 onClick={onClick}/>;
+}
+
+export function Icon({classNames, onClick = null}: IconProps) {
```
namespace me!

JanMouwes commented on src/js/components/shared/icons/loading-icons.component.jsx:
```diff
@@ -1,12 +1,26 @@
 import React from "react";
+import {FontAwesomeIcon} from "./icons.component";
 
 const LoadingIcons = {};
```
TS me then namespace me!

JanMouwes commented on src/js/components/shared/list-html/list-html.component.tsx:
```diff
@@ -0,0 +1,58 @@
+    private readonly _tagName: any;
+    private readonly _createListItem: (T) => Element;
+
+    constructor(props, tagName) {
```
Type me!

JanMouwes commented on src/js/components/shared/overlay/overlay.component.tsx:
```diff
@@ -0,0 +1,40 @@
+        foreground: JSX.Element
+    };
+    
+    constructor(props) {
```
Type me!

JanMouwes commented on src/js/controllers/localstorage/esgf-localstorage.controller.ts:
```diff
@@ -0,0 +1,32 @@
+        return Localstorage.map(this.converter.fromJSONObject);
+    }
+
+    setLocalstorage(Localstorage: T[]) {
```
why not just 'set'?

JanMouwes commented on src/js/controllers/localstorage/esgf-localstorage.controller.ts:
```diff
@@ -0,0 +1,32 @@
+        localStorage.setItem(this._storageKey, LocalstorageString);
+    }
+
+    isLocalstorageSet(): boolean {
```
why not just 'isSet'?

JanMouwes commented on src/js/data/catalog/catalog.provider.interface.ts:
```diff
@@ -0,0 +1,5 @@
+import ESGFDataNodeResultDTO from "../../model/dto/esgf-data-node-result.dto";
+
+export default interface ICatalogProvider {
```
namespace me!

JanMouwes commented on src/js/data/converters/converter.interface.ts:
```diff
@@ -0,0 +1,4 @@
+export default interface IConverter<TObject, TJSONObject> {
```
IJsonConverter would be more accurate

JanMouwes commented on src/js/data/converters/preset-converter.ts:
```diff
@@ -0,0 +1,42 @@
+import IConverter from "./converter.interface";
+import { ESGFFilterProvider } from "../esgf-filter/esgf-filter.provider";
+
+export type EsgfFilterPropertyJSONDTO = {
```
namespace me!

JanMouwes commented on src/js/data/converters/qf-tile-converter.ts:
```diff
@@ -0,0 +1,37 @@
+
+export type QFFilterTileJSONDTO = { colour:string, icon:string, title:string, properties: EsgfFilterPropertyJSONDTO[]};
+
+export class QFTileConverter implements IConverter<QFFilterTileDTO, QFFilterTileJSONDTO> {
```
namespace me!

JanMouwes commented on src/js/data/esgf-search/esgf-search-results.provider.ts:
```diff
@@ -0,0 +1,31 @@
+     *
+     * @return {Promise<ESGFSearchResultDTO>}
+     */
+    async provide(query: EsgfSearchQuery) {
```
type me!

JanMouwes commented on src/js/data/esgf-search/esgf-search-results.provider.ts:
```diff
@@ -0,0 +1,31 @@
+    /**
+     * @param {ESGFSearchService} searchService
+     */
+    constructor(searchService) {
```
type me!

JanMouwes commented on src/js/model/dto/esgf-search-result.dto.ts:
```diff
@@ -0,0 +1,38 @@
+        this._limit = limit;
+    }
+
+    getNodeResults(dataNode: string) {
```
type me!

JanMouwes commented on src/js/model/factories/info-tab-vm.factory.ts:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    createPresetVM({ title, description }: PresetDTO) {
+        return new InfoTabVM("Info - Property", title, { "Description": description });
```
"Info - Property"?

JanMouwes commented on src/js/model/factories/list-item-factory.factory.tsx:
```diff
@@ -0,0 +1,149 @@
+import ArrowIcons from "../../components/shared/icons/arrow-icons.component";
+import { PresetDTO } from "../dto/esgf-preset.dto";
+
+export default class ListItemFactoryFactory {
```
This whole class is a mess.

JanMouwes commented on src/js/model/factories/overlay.factory.tsx:
```diff
@@ -0,0 +1,16 @@
+     *
+    *@summary Creates overlay
+    */
+    createOverlay(background: JSX.Element, foreground: JSX.Element, onClick: any, index: any): JSX.Element {
```
Why is this a thing at all? It's just a wrapper. This class would only be useful if it were to implement an IOverlayFactory-interface

JanMouwes commented on src/js/model/dto/esgf-catalog-item.dto.ts:
```diff
@@ -0,0 +1,169 @@
+    }
+}
+
+const json = {
```
delete this

JanMouwes commented on src/js/model/formatters/esgf-property-dto.formatter.ts:
```diff
@@ -0,0 +1,42 @@
+/**
+ * @deprecated use (or make) a ViewModel instead
+ */
+class ESGFPropertyDTOFormatter {
```
namespace & clean me!

JanMouwes commented on src/js/searchers/esgf-filter.searcher.ts:
```diff
@@ -0,0 +1,39 @@
+import Searcher, {SearchComparator} from "./searcher";
+
+class ESGFFilterSearcher {
+    private _searcherFactory: Searcher<ESGFFilterDTO>;
```
'searcherFactory'?

JanMouwes replied:
misleading name

JanMouwes commented on src/js/searchers/esgf-property.searcher.ts:
```diff
@@ -0,0 +1,41 @@
+import ESGFFilterPropertyDTO from "../model/dto/esgf-filter-property.dto";
+import ESGFPropertyDTOFormatter from "../model/formatters/esgf-property-dto.formatter";
+
+class ESGFPropertySearcher {
```
namespace me! Searchers.Esgf.PropertySearcher

JanMouwes commented on src/js/searchers/searcher.ts:
```diff
@@ -0,0 +1,33 @@
+
+export type SearchComparator<T> = (query: string, possibleMatch: T) => boolean;
+
+class Searcher<T> {
```
namespace me!

JanMouwes commented on src/js/sorters/comparators/esgf.comparator.js:
```diff
@@ -1,4 +1,5 @@
 import {alphabeticalComparator} from "./primitive.comparator";
+import { PresetDTO } from "../../model/dto/esgf-preset.dto";
```
TS and type me!

JanMouwes commented on src/js/sorters/comparators/primitive.comparator.js:
```diff
@@ -16,4 +16,20 @@ export function alphabeticalComparator(item) {
+ *
+ * @return {ComparatorObject}
+ */
+export function alphabeticalIgnoreCaseComparator(item) {
```
namespace me!

JanMouwes commented on src/js/util/string.util.js:
```diff
@@ -29,7 +29,16 @@ export function firstToLower(string) {
-    firstToUpper: firstToUpper
-};
\ No newline at end of file
+export function trimCharsLeft(string, ...charsToTrim) {
```
namespace me!

JanMouwes committed:
- Fixed synchronisation issue

marijnkoel committed:
Update result-dataset-list-item.component.tsx

marijnkoel commented on src/js/components/shared/overlay/overlay.component.tsx:
```diff
@@ -0,0 +1,40 @@
+
+        return (
+            <div className="overlayContainer">
+                <div className="background">{background}</div>
```
still dont like this idk why

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291566940

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291571022

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291573391

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291575015

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291576724

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291580252

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291580618

marijnkoel commented on src/js/managers/esgf-search.manager.ts:
```diff
@@ -0,0 +1,56 @@
+    private readonly _eventEmitter: any;
+    private readonly _eventNames: { searched, searchStarted } = {searched: "searched", searchStarted: "search-started"};
+
+    private _currentQuery: EsgfSearchQuery;
```
esgf?

marijnkoel commented on src/js/model/formatters/esgf-property-dto.formatter.ts:
```diff
@@ -0,0 +1,42 @@
+     * @param {ESGFFilterPropertyDTO} property
+     * @return {string}
+     */
+    static toHumanText(property) {
```
that name though

JanMouwes committed:
- Added catalogue-tests

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291615181
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291614574

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291615453

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291620204

JanMouwes committed:
- Fixed comment https://github.com/christiaanko/ESGF_Portal_React/pull/18#discussion_r291621578
